### ISSUE TITLE
Slow file saving

### ISSUE THREAD

**Issue by BlueTeo91** [microsoft/vscode#76470]
Issue Type: <b>Performance Issue</b>

The file saving requires few seconds from the last VSCode update

VS Code version: Code 1.35.1 (c7d83e57cd18f18026a8162d042843bda1bcf21f, 2019-06-12T14:30:02.622Z)
OS version: Windows_NT x64 10.0.16299

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-8650U CPU @ 1.90GHz (8 x 2112)|
|GPU Status|2d_canvas: disabled_software<br>checker_imaging: disabled_off<br>flash_3d: disabled_software<br>flash_stage3d: disabled_software<br>flash_stage3d_baseline: disabled_software<br>gpu_compositing: disabled_software<br>multiple_raster_threads: disabled_off<br>native_gpu_memory_buffers: disabled_software<br>rasterization: disabled_software<br>surface_synchronization: disabled_off<br>video_decode: disabled_software<br>webgl: disabled_off<br>webgl2: disabled_off|
|Load (avg)|undefined|
|Memory (System)|15.88GB (7.71GB free)|
|Process Argv||
|Screen Reader|no|
|VM|0%|
</details><details>
<summary>Process Info</summary>

```
CPU %	Mem MB	   PID	Process
    5	   123	 14408	code main
    1	    61	  6668	   window (Issue Reporter)
    0	    38	 11004	   shared-process
    0	   199	 16924	   window (PHY_abstraction_SERIAL.m - esim2m - Visual Studio Code)
    0	    23	 12040	     extensionHost
    0	     4	 12820	     watcherService 
    0	     1	 12144	       console-window-host (Windows internal process)
```

</details>
<details>
<summary>Workspace Info</summary>

```
|  Window (PHY_abstraction_SERIAL.m - esim2m - Visual Studio Code)
|    Folder (esim2m): 2426 files
|      File types: png(1299) mat(620) fig(155) txt(112) m(94) pdf(28) eps(13)
|                  mexw64(12) ant(12) xlsx(10)
|      Conf files:;
```

</details>
<details><summary>Extensions (8)</summary>

Extension|Author (truncated)|Version
---|---|---
matlab-formatter|Aff|2.4.12
Bookmarks|ale|10.4.4
notes|can|0.9.3
matlab|Gim|1.1.0
latex-workshop|Jam|7.0.2
matlab-snippets|sky|1.11.0
matlab-complete|sla|1.0.2
markdown-all-in-one|yzh|2.4.0

(1 theme extensions excluded)

</details>

**Comment by bpasero** [comment]
I wonder if this issue is being caused by an installed extension. Can you try to run VS Code without extensions? From the command line (NOT the integrated terminal in Code), execute: `code --disable-extensions` and try your steps again to see if it reproduces. If you see it is an issue with the extension, please file it against the extension repository itself.

**Comment by BlueTeo91** [comment]
I tried, the saving time seems to be back to normal but what I didn't get is why the extensions are still active when I open VScode.
Furthermore what probably gives problem is the builtin git extension, which doesn't work properly since the last update

**Comment by bpasero** [comment]
@BlueTeo91 which extension is still active? Can you try to disable extensions one by one to find the one in charge of this issue?

**Comment by BlueTeo91** [comment]
I'm trying disabling extension by extension... Most likely the problem is with the builtin git extension, but I still have to try disable some of the other extensions just to confirm it

**Comment by bpasero** [comment]
Please do.

**Comment by BlueTeo91** [comment]
I tried to disable all the extensions trying them one by one, but apparently after VSCode update of today the problem does not show up anymore

**Comment by BlueTeo91** [comment]
Unfortunately the problem is still there...
Already tried to disable one by one all the extensions, still the problem appears randomly. 
I also tried to disable the builtin git extension because on top of the slow saving, when the problem appears, also the git extension stop working as well as the bookmarks one.

At the beginning I thought the problem could have been the bookmarks or git extensions causing the problem, but even with those disabled (simultaneously or one at a time), the problem shows up.

Those are the extensions I installed:
![Capture](https://user-images.githubusercontent.com/2975798/60710843-de0b7a80-9f13-11e9-8704-f64620243884.PNG)
